### ISSUE TITLE
No syntax error reported on tripple-quoted f-string with backslash

### ISSUE THREAD

**Issue by Tomaz-Vieira** [microsoft/pyright#4552]
**Describe the bug**
Pyright does not report a syntax error on tripple-quoted f-strings that have a backslash inside them.

**To Reproduce**
Running this code:

```python
#pyright: strict
x = f"""
    {"\n".join(['a', 'b'])}
"""
```

errors with:

```
SyntaxError: f-string expression part cannot include a backslash
```

but no error is reported by pyright.

**Expected behavior**
Error report pointing to the usage of the backslash

**VS Code extension or command-line**
pyright extension version 1.1.291

**Comment by erictraut** [comment]
Thanks for the bug report. Pyright already contained a check for this condition, but there was a bug in the logic that caused the error to go unreported.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.292, which I just published. It will also be included in a future release of pylance.
